src/libs/bin_decoder/lv_bin_decoder.c

```diff
@@ -1008,42 +864,149 @@ static lv_result_t decompress_image(lv_image_decoder_dsc_t * dsc, const lv_image
     return LV_RESULT_OK;
 }
 
-static lv_result_t try_cache(lv_image_decoder_dsc_t * dsc)
+static bool bin_decoder_decode_data(lv_image_decoder_dsc_t * dsc)
 {
-    lv_cache_lock();
+    lv_image_decoder_t * decoder = dsc->decoder;
+
+    lv_fs_res_t res = LV_RESULT_INVALID;
+    // uint32_t t = lv_tick_get();
```

Comment:
So current cache interface lacks of `weight`. It should be a parameter of `lv_cache_add` as discussed on meeting.

Reply:
I have an idea about this, but we can't use this feature now. I'll add it back later.